src/WebJobs.Script/Host/HostFunctionMetadataProvider.cs

```diff
@@ -179,10 +183,21 @@ internal static string ParseLanguage(string scriptFilePath, IEnumerable<RpcWorke
 
             // determine the script type based on the primary script file extension
             string extension = Path.GetExtension(scriptFilePath).ToLowerInvariant().TrimStart('.');
-            var workerConfig = workerConfigs.FirstOrDefault(config => config.Description.Extensions.Contains("." + extension));
-            if (workerConfig != null)
+
+            // In the special case of a null FUNCTIONS_WORKER_RUNTIME and an extension of "dll",
+            // we want to skip the worker config check (to prevent the dotnet-isolated worker from
+            // being chosen) and allow in-proc handling to be used.
+            // Other notes:
+            // - when dotnet-isolated, the FUNCTIONS_WORKER_RUNTIME is required to be set to "dotnet-isolated"
+            // - when FUNCTIONS_WORKER_RUNTIME is set to "dotnet", workerConfigs is empty so this correctly falls
+            //   through to choose DotNetAssembly with a "dll" extension
+            if (!(functionsWorkerRuntime is null && extension == "dll"))
             {
-                return workerConfig.Description.Language;
+                var workerConfig = workerConfigs.FirstOrDefault(config => config.Description.Extensions.Contains("." + extension));
```

Comment:
I realize this is pre-existing code, and wouldn't expect this would ever be the case (and have no concrete scenario used today), but the current check could potentially lead to a false positive on a partial match (e.g., `.js` matching on `.jsx`).

Would also be good to make sure the check here is case insensitive.

Reply:
Extensions is a `List<string>` so it's doing a direct comparison to strings in the list -- so .jsx won't match with .js.

However, I will add `OrdinalIgnoreCase` to this check.